docs/source/api/plugin/subscription-callback.mdx

```diff
@@ -0,0 +1,122 @@
+---
+title: "API Reference: Subscription callback plugin"
+api_reference: true
+minVersion: 4.9.0
+---
+
+> This protocol is currently in [preview](/resources/product-launch-stages#preview). Breaking changes might be introduced during the preview period.
+>
+> This document covers the usage of the Subscription callback plugin for use in Federation with Apollo Router. For more information about the protocol itself, see the [subscription callback protocol](/router/executing-operations/subscription-callback-protocol).
+
+## Using the plugin
+
+This article documents the options for the `ApolloServerPluginSubscriptionCallback` plugin, which you can import from `@apollo/server/plugin/subscriptionCallback`.
+
+This plugin enables your GraphQL Server to respond to subscription operations using the subscription callback protocol. This protocol is used by Apollo Router to execute subscription operations and receive updates at a URL specified by the router.
+
+This feature can only be enabled by providing an `ApolloServerPluginSubscriptionCallback` instance to your `ApolloServer` constructor:
+
+<MultiCodeBlock>
+
+```ts
+import { ApolloServer } from '@apollo/server';
+import { ApolloServerPluginSubscriptionCallback } from '@apollo/server/plugin/subscriptionCallback';
+
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
+  plugins: [
+    ApolloServerPluginSubscriptionCallback(),
+  ],
+});
+```
+
+</MultiCodeBlock>
+
+## Caveats
+
+The subscription plugin implementation inherently bypasses Apollo Server's request lifecycle. This means that certain plugin hooks (notably `executionDidStart` and `willResolveField`) will not be called when handling callback subscription requests or when sending subscription events. There is currently no metrics or tracing support for callback subscriptions.
+
+#### Options
+
+<table class="field-table">
+  <thead>
+    <tr>
+      <th>Name /<br/>Type</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+
+<tbody>
+
+<tr>
+<td>
+
+###### `heartbeatIntervalMs`
+
+`number`
+</td>
+<td>
+
+Optionally configure the heartbeat interval (provided in milliseconds). The default is 5 seconds, which is the interval that Apollo Router expects. Lengthening this interval will cause Apollo Router to invalidate existing subscriptions frequently and is not recommended. You might want to shorten this interval if you have latency issues between your GraphQL Server and Apollo Router.
+
+</td>
+</tr>
+
+<tr>
+<td>
+
+###### `logger`
+
+`Logger`
+</td>
+<td>
+
+Optionally provide a `Logger` instance to capture logs from the plugin.
```

Comment:
Is there a good link we can add to `Logger` in case folks are unfamiliar?

Reply:
Yep, good call! Added